### ISSUE TITLE
CqlClientConnection.Close() blocked forever

### ISSUE THREAD

**Issue by joao-r-reis** [datastax/go-cassandra-native-protocol#55]
I believe this issue is a bit different from #36 and #38 because the blocked channel here is not the same as #38 .
`Close()` sets `outgoing` channel to `nil` and in the `outgoingLoop` if the channel becomes `nil` just before the loop tries to read from that channel, it will be blocked forever. 

There's a check to see if the connection is closed just before that read but there's no locking involved so I believe this issue can occur. I saw this happen in a Github Action so I couldn't see exactly where the goroutines were blocked but looking at the logs I believe this is what happened.

**Comment by adutra** [comment]
Yes this looks different. It would be good to tackle #50 so that we can better isolate all the race issues in the client package.
